src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java

```diff
@@ -150,7 +128,7 @@ public String getURI() {
      */
     @Override
     public boolean hasChildNodes() {
-        return !children.isEmpty();
+        return !getChildren().isEmpty();
```

Comment:
`getChildren()` always create a collection now.
can we avoid this creation in internal implementations ?

Reply:
done